packages/grid/_modules_/grid/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -33,12 +33,18 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridCellPa
 
     const handleRef = useForkRef(checkboxElement, ref);
     const element = apiRef.current.getCellElement(id, field);
+    const shiftRef = React.useRef(false);
 
     const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-      apiRef.current.selectRow(id, event.target.checked, false);
+      if (shiftRef.current) {
+        apiRef.current.expandRowRangeSelection(id);
```

Comment:
I don't get why we should expose this method in the public API. You could listen for the `cellClick` event in `useGridSelection` and check if the clicked cell is the one with the checkbox.

Reply:
We stop the propagation in the checkbox component.
Probably to avoid double call when clicking on the checkbox
And if we just keep the cell click and remove the checkbox, then it would not work with the other types of interaction that triggers the `onChange`.
We could replicate them (the `Enter` keydown for instance) but I don't think it's a good pattern.

Reply:
We could also remove the `stopPropagation` from the checkbox `onClick` handler and prevent the double call in another way. But my ideas to do it are hacky.

Reply:
> Add a rowSelectionCheckboxChange event and just fire the event in the checkbox component, all the behavior could then be handled in useGridSelection

This one sounds better. It avoids adding another method to the API and keeps everything related to selection in one place.

Reply:
PR updated with an event-based implementation